tests/kafkatest/sanity_checks/test_mirror_maker.py

```diff
@@ -0,0 +1,98 @@
+        self.zk2 = ZookeeperService(test_context, num_nodes=1)
+
+        self.k1 = KafkaService(test_context, num_nodes=1, zk=self.zk1,
+                                  topics={self.topic: {"partitions": 1, "replication-factor": 1}})
+        self.k2 = KafkaService(test_context, num_nodes=1, zk=self.zk2,
+                                  topics={self.topic: {"partitions": 1, "replication-factor": 1}})
+
+        self.num_messages = 1000
+        # This will produce to source kafka cluster
+        self.producer = VerifiableProducer(test_context, num_nodes=1, kafka=self.k1, topic=self.topic,
+                                           max_messages=self.num_messages, throughput=1000)
+        self.mirror_maker = MirrorMaker(test_context, sources=[self.k1], target=self.k2, whitelist=self.topic)
+
+        # This will consume from target kafka cluster
+        self.consumer = ConsoleConsumer(test_context, num_nodes=1, kafka=self.k2, topic=self.topic,
+                                        consumer_timeout_ms=10000)
+
+    def setUp(self):
+        # Source cluster
+        self.zk1.start()
+        self.k1.start()
+
+        # Target cluster
+        self.zk2.start()
+        self.k2.start()
+
+    def test_lifecycle(self):
+        """Start and stop a single-node MirrorMaker and validate that the process appears and disappears in a
+        reasonable amount of time.
+        """
+        self.mirror_maker.start()
+        node = self.mirror_maker.nodes[0]
+        wait_until(lambda: self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took too long to start.")
+
+        self.mirror_maker.stop()
+        wait_until(lambda: not self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took to long to stop.")
+
+    def test_end_to_end(self):
+        """
+        Test end-to-end behavior under non-failure conditions.
+
+        Setup: two single node Kafka clusters, each connected to its own single node zookeeper cluster.
+        One is source, and the other is target.
+
+        - Start mirror maker.
+        - Produce a small number of messages to the source cluster.
+        - Consume messages from target.
+        - Confirm that number of consumed messages matches the number produced.
+        """
+        self.mirror_maker.start()
+        node = self.mirror_maker.nodes[0]
+        wait_until(lambda: self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took too long to start.")
+
+        self.producer.start()
+        self.producer.wait()
+        self.consumer.start()
+        self.consumer.wait()
+        assert len(self.consumer.messages_consumed[1]) == self.num_messages
+
+        self.mirror_maker.stop()
+        wait_until(lambda: not self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took to long to stop.")
```

Comment:
Hm, not sure what to say here other than the ever-unhelpful "the tests are passing on my machine"

This may be a process cleanup issue as we discussed offline. I added a message in test_end_to_end to clarify how many produced/consumed to make it more obvious whether the error is legit